charts/nri-statsd/templates/_helpers.tpl

```diff
@@ -78,14 +78,18 @@ Define the statsd metric tags
 Return the insightsKey
 */}}
 {{- define "nri-statsd.insightsKey" -}}
-{{- if .Values.global}}
-  {{- if .Values.global.insightsKey }}
-      {{- .Values.global.insightsKey -}}
-  {{- else -}}
-      {{- .Values.insightsKey | default "" -}}
-  {{- end -}}
+{{- if or (include "newrelic.common.license._customSecretName" .) (include "newrelic.common.license._customSecretKey" .) -}}
```

Comment:
You are using internal functions from the `common-library` that should not be used: https://github.com/newrelic/helm-charts/blob/bda20f2057e742cede14dd947ccb579f27993fb0/library/common-library/templates/_license.tpl#L29-L32

The point of using these functions is to set which key inside the secret will be used whenever the case the key is created by the user of by the values file.

You would need to create a new function in the `common-library` that supports `insightsKey` and you should not use the internal ones in the final chart but the final ones directly in the deployment. Check the usage here: https://github.com/newrelic/helm-charts/blob/bda20f2057e742cede14dd947ccb579f27993fb0/library/common-library/DEVELOPERS.md#_licensetpl

Reply:
@kang-makes Although the `common-library` does not support the `insightsKey`, this essentially just a licenseKey I believe? Is there any way we could just use that functionality to implement it?

Reply:
The way we support installation on Kubernetes is by using `nri-bundle`, not individual charts: https://github.com/newrelic/helm-charts/blob/bda20f2057e742cede14dd947ccb579f27993fb0/charts/nri-bundle/Chart.yaml

I know that `nri-statsd` is an exception but we work in a way that charts are composable and having the same UX so we should support them in a consistent way. Hence the `common-library` creating a seamless UX though all the charts.

If we are going to support a new global key we should support it in the `common-library` so other charts can use it.

Reply:
Understood, well we would love to make use of this feature to ensure that we are not hardcoding API keys into our source control using the existing `nri-statsd` chart. Our goal is to inject the secret into the cluster before referencing the secret name in the chart